packages/ics721/src/execute.rs

```diff
@@ -623,6 +643,21 @@ where
         if let Some(collection_data) = collection_data {
             instantiate_msg.name = collection_data.name;
             instantiate_msg.symbol = collection_data.symbol;
+            let collection_info_extension_msg =
+                collection_data
+                    .extension
+                    .map(|ext| cw721::msg::CollectionExtensionMsg {
+                        description: Some(ext.description),
+                        image: Some(ext.image),
+                        external_link: ext.external_link,
+                        explicit_content: ext.explicit_content,
+                        start_trading_time: ext.start_trading_time,
+                        royalty_info: ext.royalty_info.map(|r| RoyaltyInfoResponse {
+                            payment_address: cw721_admin_or_ics721_admin_or_ics721_creator, // r.payment_address cant be used, since it is from another chain
+                            share: r.share,
+                        }),
+                    });
+            instantiate_msg.collection_info_extension = collection_info_extension_msg;
```

Comment:
on ibc receive for initial transfer, collection extension is passed to cw721 instantiate message.